### ISSUE TITLE
Remove dubious usage of pyOpenSSL's netscape in twisted/internet/_sslverify.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8441]
|[<img alt="alex's avatar" src="https://avatars.githubusercontent.com/u/772?s=50" width="50" height="50">](https://github.com/alex)| @alex reported|
|-|-|
|Trac ID|trac#8441|
|Type|enhancement|
|Created|2016-06-09 12:32:18Z|

<details><summary>Searchable metadata</summary>

```
trac-id__8441 8441
type__enhancement enhancement
reporter__alex alex
priority__normal normal
milestone__None None
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1465475538567182 1465475538567182
changetime__1469316197571655 1469316197571655
version__None None
owner__glyph glyph
cc__alex
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

In review as per https://github.com/twisted/twisted/pull/146

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban set owner to @glyph|
|-|-|

Changes look good.

I think that the new pyOpenSSL dependence should be announce in the release notes. I don't see any update on the wiki page dedicated to Release Manager Notes  https://twistedmatrix.com/trac/wiki/ReleaseManagerNotes

Please add a note there.

Maybe this should be a .feature topfile to announce that we now depend on pyOpenSSL 16.0.0 ... in case we forget to announce this as part of the top paragraph in the release notes.

----------

Please also create a branch in Twisted repo and send it to buildbot.

As a Twisted overlord you can just merge this without running the buildbot tests ... but I think that is better to have those tests executed before the merge.... I don't expect anything to break, but you never know

Please check my comments and merge.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph commented|
|-|-|

For posterity, we did have a mailing list discussion about the compatibility implications of this version update, and I think the consensus was that we should proceed: https://twistedmatrix.com/pipermail/twisted-python/2016-July/030536.html

Rather than making a dependency update a special release-process exception thing though, I think it would make more sense to just include it in a topfile so it is included with everything else.  Perhaps later we'll want to add a new topfile type.

Alex, would you mind changing the `.misc` to a ... `.removal` I guess, since the interesting thing to users is that we are dropping support for older pyOpenSSL and Cryptography versions?  (If not I will probably make the change myself and land regardless :).)

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph commented|
|-|-|

Alex took care of the feedback and I sent the PR to buildbot, so as soon as those builds are done we will land.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|GitHub set status to `closed`|
|-|-|

In changeset 2e8bcb3073e77f15d6c04bf601c57f653b199a2b
```
#!CommitTicketReference repository="" revision="2e8bcb3073e77f15d6c04bf601c57f653b199a2b"
Merge pull request #146 from alex/patch-2

Author: alex

Reviewer: glyph, adiroiban

Fixes: #8441

Remove use of deprecated, to-be-removed Netscape SPKI functions, and instead, determine the hash of the public key by actually extracting the public key bytes and then hashing them.  This gives the same values as the method did previously (since this is what the Netscape methods were doing).

Additionally, bump the minimum required version of pyOpenSSL to 16.0 and inform users that this is the case via the changelog.
```
